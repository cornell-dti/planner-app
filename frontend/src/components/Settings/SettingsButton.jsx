// @flow strict

import * as React from 'react';
import TagColorConfigEditor from '../ColorConfigEditor/ColorConfigEditor';
import styles from './SettingsButtonStyles.css';

type State = {|
  +showSettings: boolean;
|};

export default class SettingsButton extends React.Component<{}, State> {
  constructor() {
    super();
    this.state = { showSettings: false };
  }

  displayModal = () => {
    this.setState({ showSettings: true });
  };

  closeModal = () => {
    this.setState({ showSettings: false });
  };

  render() {
    const { showSettings } = this.state;
    const showStyles = showSettings ? { display: 'block' } : { display: 'none' };
    return (
      <div style={{ display: 'inline-block' }}>
        <button type="submit" onClick={this.displayModal}>Settings</button>
        <div className={styles.settingsModal}>
          <div style={showStyles}>
            <button className={styles.closeButton} type="submit" onClick={this.closeModal}>
              Close
            </button>
            <TagColorConfigEditor />
          </div>
        </div>
      </div>
    );
  }
}
