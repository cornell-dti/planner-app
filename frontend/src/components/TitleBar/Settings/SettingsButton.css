.CloseButton {
  position: absolute;
  top: 40px;
  right: 40px;
  color: white;
  font-size: 30px;
}

.ChangesSaved {
  position: absolute;
  top: 40px;
  right: 100px;
  line-height: 30px;
  padding: 0;
  margin: 0;
  font-style: italic;
}

.SettingsModal {
  position: fixed;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 10;
  background-color: #7b7b7b;
  color: white;
  overflow-y: auto;
}

.SettingsButton {
  width: 17.5px;
  height: 17.5px;
}

.ContentWrap {
  padding: 60px 50px;
}
