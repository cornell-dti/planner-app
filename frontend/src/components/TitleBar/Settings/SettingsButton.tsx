import React, { ReactElement } from 'react';
import SettingsPage from './SettingsPage';
import styles from './SettingsButton.module.css';
import SamwiseIcon from '../../UI/SamwiseIcon';

export default function SettingsButton(): ReactElement {
  const [showSettings, setShowSettings] = React.useState(false);

  const displayModal = (): void => setShowSettings(true);
  const closeModal = (): void => setShowSettings(false);

  return (
    <div style={{ display: 'inline-block' }}>
      <button type="submit" onClick={displayModal}>
        <p style={{ transform: 'scale(2)translateY(-5px)' }} title="Settings Button"><SamwiseIcon iconName="settings" /></p>
      </button>
      {showSettings && (
        <div className={styles.SettingsModal}>
          <button className={styles.CloseButton} type="submit" title="Close Settings" onClick={closeModal} tabIndex={-1}>
            <SamwiseIcon iconName="x-light-settings" />
          </button>
          <section className={styles.ContentWrap}>
            <h2>Settings</h2>
            <SettingsPage />
          </section>
        </div>
      )}
    </div>
  );
}
