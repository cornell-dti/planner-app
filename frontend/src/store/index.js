// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';

const store: Store<State, $Subtype<Action>> = createStore(rootReducer);
export default store;
