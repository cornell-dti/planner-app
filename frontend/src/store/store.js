// @flow strict

import { createStore } from 'redux';
import type { Store } from 'redux';
import rootReducer from './reducers';
import type { State } from './store-types';
import type { Action } from './action-types';
import type { AppUser } from '../util/firebase-util';
import { error } from '../util/general-util';

export type GlobalStore = Store<State, $Subtype<Action>>;

const store: GlobalStore = createStore(rootReducer);
let appUser: AppUser | null = null;

/**
 * Initialize the app and record a global user and returns the initialized store.
 *
 * @param {AppUser} user the user of the app.
 * @return {GlobalStore} the newly initialized store.
 */
export function initializeApp(user: AppUser): GlobalStore {
  appUser = user;
  return store;
}

/**
 * Returns the global app user.
 *
 * @return {AppUser} the global app user.
 */
export const getAppUser = (): AppUser => appUser ?? error('App is not initialized.');

/**
 * Dispatch an action and returns the dispatched action.
 *
 * @param {Action} action the action to dispatch.
 * @return {Action} the action to dispatch.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
