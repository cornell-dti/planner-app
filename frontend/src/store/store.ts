import { createStore, Store } from 'redux';
import rootReducer from './reducers';
import { State } from './store-types';
import { Action } from './action-types';
import { initialStateForTesting } from './state';

export type GlobalStore = Store<State, Action>;

/**
 * The redux store. It should only be given to the react-redux provider.
 */
export const store: GlobalStore = createStore(rootReducer);

/**
 * The redux store for testing.
 */
export const storeForTesting: GlobalStore = createStore(
  (initialState: State = initialStateForTesting, action: Action): State => rootReducer(
    initialState, action,
  ),
);

/**
 * Dispatch an action and returns the same dispatched action.
 */
export const dispatchAction = (action: Action): Action => store.dispatch(action);
