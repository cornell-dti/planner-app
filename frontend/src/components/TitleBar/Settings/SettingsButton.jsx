// @flow strict

import React from 'react';
import Settings from '../../../assets/svgs/settings.svg';
import Delete from '../../../assets/svgs/X.svg';
import SettingsPage from './SettingsPage';
import styles from './SettingsButton.css';

export default function SettingsButton() {
  const [showSettings, setShowSettings] = React.useState(false);

  const displayModal = () => setShowSettings(true);
  const closeModal = () => setShowSettings(false);

  // /**
  //  * Display the settings model.
  //  */
  // displayModal = () => this.setState({ showSettings: true });
  //
  // /**
  //  * Close the settings model.
  //  */
  // closeModal = () => this.setState({ showSettings: false });
  return (
    <div style={{ display: 'inline-block' }}>
      <button type="submit" onClick={displayModal}>
        <span style={{ backgroundImage: `url(${Settings})` }} />
      </button>
      {showSettings && (
        <div className={styles.settingsModal}>
          <button className={styles.closeButton} type="submit" onClick={closeModal}>
            <span style={{ backgroundImage: `url(${Delete})` }} />
          </button>
          <section className={styles.contentWrap}>
            <h2>Settings</h2>
            <SettingsPage />
          </section>
        </div>
      )}
    </div>
  );
}
