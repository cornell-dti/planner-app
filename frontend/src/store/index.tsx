import React, { ReactElement } from 'react';
import { Provider as ReactReduxProvider } from 'react-redux';
import { store, storeForTesting } from './store';

type Props = { readonly children: ReactElement };

export const Provider = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={store}>{children}</ReactReduxProvider>
);

export const ProviderForTesting = ({ children }: Props): ReactElement => (
  <ReactReduxProvider store={storeForTesting}>{children}</ReactReduxProvider>
);
