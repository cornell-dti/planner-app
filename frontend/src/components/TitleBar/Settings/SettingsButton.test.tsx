import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from 'store/store';
import SettingsButton from './SettingsButton';

it('SettingsButton matches snapshot.', () => {
  const tree = renderer.create(<Provider store={store}><SettingsButton /></Provider>).toJSON();
  expect(tree).toMatchSnapshot();
});
