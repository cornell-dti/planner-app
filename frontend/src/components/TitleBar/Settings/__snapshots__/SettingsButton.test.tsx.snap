// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsButton matches snapshot. 1`] = `
<div
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <button
    className="SettingsButton"
    onClick={[Function]}
    type="submit"
  >
    <p
      style={
        Object {
          "transform": "scale(2)translateY(-5px)",
        }
      }
      title="Settings Button"
    >
      <span
        title="Settings"
      >
        <svg
          alt="Settings"
          height="1em"
          onKeyUp={[Function]}
          tabIndex={0}
          title="Settings"
          width="1em"
        >
          settings.svg
        </svg>
      </span>
    </p>
  </button>
</div>
`;
